From 8e8ba024acaa9f09b43a9a96a60031aaabfaba5e Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Tue, 4 Jun 2019 18:31:58 +0200
Subject: GCC: XSetWindowAttributes struct initialization should keep order of
 declaration

XSetWindowAttributes initialization of attributes in GLSurfaceGLX is not in the
same order of the declaration. GCC fails because of that.

Bug: 819294
Change-Id: I8a97da980d5961a35a47ae4d0d8d558b85291f1f

diff --git a/ui/gl/gl_surface_glx.cc b/ui/gl/gl_surface_glx.cc
index f649dd469bac..0aa6892014f2 100644
--- a/ui/gl/gl_surface_glx.cc
+++ b/ui/gl/gl_surface_glx.cc
@@ -583,10 +583,10 @@ bool NativeViewGLSurfaceGLX::Initialize(GLSurfaceFormat format) {
 
   XSetWindowAttributes swa = {
       .background_pixmap = 0,
-      .bit_gravity = NorthWestGravity,
-      .colormap = g_colormap,
       .background_pixel = 0,  // ARGB(0,0,0,0) for compositing WM
       .border_pixel = 0,
+      .bit_gravity = NorthWestGravity,
+      .colormap = g_colormap,
   };
   auto value_mask = CWBackPixmap | CWBitGravity | CWColormap | CWBorderPixel;
   if (ui::IsCompositingManagerPresent() &&
-- 
2.20.1

